src/material-examples/dialog-data/dialog-data-example.ts

```diff
@@ -0,0 +1,29 @@
+import {Component, Inject} from '@angular/core';
+import {MdDialog, MD_DIALOG_DATA} from '@angular/material';
+
+/**
+ * @title Dialog Data Sharing
+ */
+@Component({
+  selector: 'dialog-data-example',
+  templateUrl: 'dialog-data-example.html',
+})
+export class DialogDataExample {
+  constructor(public dialog: MdDialog) {}
+
+  openDialog() {
+    this.dialog.open(DialogDataExampleDialog, {
+      data: {
+        animal: 'panda'
```

Comment:
This data should come from an `<input>` next to the dialog open button, to demonstrate the value coming from the user.